#! /usr/bin/env python


# Initialize ros and import parameter generator
PACKAGE='robot_shared_control'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# Add parameters
#       Name     Type    Level Description     	Default 	Min   	Max
gen.add("filteredForceGain",double_t,0,"Filtering weight of force torque sensor",0.9,0,1)
gen.add("velocityLimit",double_t,0,"Velocity limit [m/s]",0.3,0.3,1)
gen.add("duration",double_t,0,"Duration [s]",60,10,180)
gen.add("kxy", double_t,0,"Stiffness along x-y direction",0,0,500)
gen.add("dxy", double_t,0,"Damping along x-y direction",0,0,10)
gen.add("kphi", double_t,0,"Stiffness around phi",0,0,0.1)
gen.add("dphi", double_t,0,"Damping around phi",0,0,1e-3)
gen.add("xyPositionMapping", double_t,0,"xy Position mapping",0.7,0.2,1)
gen.add("zPositionMapping", double_t,0,"z Position mapping",0.5,0.2,1)

exit(gen.generate(PACKAGE, "robot_shared_control", "footControl_params"))